src/adr/Utils/FileUtils.cs

```diff
@@ -0,0 +1,136 @@
+﻿using System.IO;
+using System.Linq;
+using System.Text.RegularExpressions;
+
+namespace adr.Utils
+{
+    /// <summary>
+    /// File management utilities
+    /// </summary>
+    internal static class FileUtils
+    {
+        /// <summary>
+        /// Sanitize a file name from invalid characters and reserved words
+        /// </summary>
+        /// <param name="filename">the file name</param>
+        /// <param name="substitution">the substitution character for invalid characters</param>
+        /// <returns>a clean file name</returns>
+        internal static string SanitizeFilename(string filename, char substitution = '_')
+        {
+            filename = filename.Replace(' ', substitution);
+            string invalidCharacters = Regex.Escape(new string(System.IO.Path.GetInvalidFileNameChars()));
+            string regex = string.Format(@"([{0}]*\.+$)|([{0}]+)", invalidCharacters);
+
+            var cleanFilename = Regex.Replace(filename, regex, substitution.ToString());
+
+            var reservedWords = new[]
```

Comment:
Regarding these reserved words: I would imagine simply disallowing path serarators would be sufficient, and then always creating them inside of the given adr path?

Reply:
yep, probably this control is excessive, also given that the structure of the file name is forced in something like `<number>_<title>.md`

the invalid file name characters should be strong enough to ensure a valid path in all os/file systems